src/cli/project/export/pypi_requirements.rs

```diff
@@ -0,0 +1,323 @@
+}
+
+#[derive(Debug)]
+struct PypiPackageReqData {
+    source: String,
+    hash_flag: Option<String>,
+    editable: bool,
+}
+
+impl PypiPackageReqData {
+    fn from_pypi_package(p: &PypiPackage) -> Self {
+        // pip --verify-hashes does not accept hashes for local files
+        let (s, include_hash) = match p.url() {
+            UrlOrPath::Url(url) => (url.as_str(), true),
+            UrlOrPath::Path(path) => (
+                path.as_os_str()
+                    .to_str()
+                    .unwrap_or_else(|| panic!("Could not convert {:?} to str", path)),
+                false,
+            ),
+        };
+        //
+        // remove "direct+ since not valid for pip urls"
+        let s = s.trim_start_matches("direct+");
+
+        let hash_flag = if include_hash {
+            get_pypi_hash_str(p.data().package)
+        } else {
+            None
+        };
+
+        Self {
+            source: s.to_string(),
+            hash_flag,
+            editable: p.is_editable(),
+        }
+    }
+
+    fn to_req_entry(&self) -> String {
+        let mut entry = String::new();
+
+        if self.editable {
+            entry.push_str("-e ");
+        }
+        entry.push_str(&self.source);
+
+        if let Some(hash) = &self.hash_flag {
+            entry.push_str(&format!(" {}", hash));
+        }
+
+        entry
+    }
+}
+
+fn get_pypi_hash_str(package_data: &PypiPackageData) -> Option<String> {
+    if let Some(hashes) = &package_data.hash {
+        let h = match hashes {
+            PackageHashes::Sha256(h) => format!("--hash=sha256:{:x}", h).to_string(),
+            PackageHashes::Md5Sha256(_, h) => format!("--hash=sha256:{:x}", h).to_string(),
+            PackageHashes::Md5(h) => format!("--hash=md5:{:x}", h).to_string(),
+        };
+        Some(h)
+    } else {
+        None
+    }
+}
+
+fn render_pypi_requirements(
+    target: impl AsRef<Path>,
+    packages: &[PypiPackageReqData],
+) -> miette::Result<()> {
+    if packages.is_empty() {
+        return Ok(());
+    }
+
+    let target = target.as_ref();
+
+    let reqs = packages
+        .iter()
+        .map(|p| p.to_req_entry())
+        .collect::<Vec<_>>()
+        .join("\n");
+
+    fs::write(target, reqs)
+        .into_diagnostic()
+        .with_context(|| format!("failed to write requirements file: {}", target.display()))?;
+
+    Ok(())
+}
+
+fn render_env_platform(
+    output_dir: &Path,
+    env_name: &str,
+    env: &Environment,
+    platform: &Platform,
+    split_nohash: bool,
+) -> miette::Result<()> {
+    let packages = env.packages(*platform).ok_or(miette::miette!(
+        "platform '{platform}' not found for env {}",
+        env_name,
+    ))?;
+
+    let mut pypi_packages: Vec<PypiPackageReqData> = Vec::new();
+
+    for package in packages {
+        match package {
+            Package::Pypi(p) => pypi_packages.push(PypiPackageReqData::from_pypi_package(&p)),
+            Package::Conda(cp) => {
+                tracing::warn!(
```

Comment:
I don't think this warning is required as it is kind of intended. We should make it clear in the help message and documentation.